vercel/resource_deployment.go

```diff
@@ -308,7 +308,26 @@ func (r resourceDeployment) Update(ctx context.Context, req tfsdk.UpdateResource
 // Delete does nothing other than clear the existing terraform state for a Deployment.
 // This is done intentionally, as typically, Vercel users do not continually delete old Deployments.
 func (r resourceDeployment) Delete(ctx context.Context, req tfsdk.DeleteResourceRequest, resp *tfsdk.DeleteResourceResponse) {
-	tflog.Trace(ctx, "deleted deployment")
+	//err := r.p.client.DeleteProject(ctx, state.ID.Value, state.TeamID.Value)
+	var state Deployment
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	dResp, err := r.p.client.DeleteDeployment(ctx, state.URL.Value, state.TeamID.Value)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error deleting deployment",
+			fmt.Sprintf(
+				"Could not delete deployment %s, unexpected error: %s",
+				state.URL.Value,
+				err,
+			),
+		)
+		return
+	}
+	tflog.Trace(ctx, fmt.Sprintf("deleted deployment %s", dResp.UID))
```

Comment:
Prefer structured logging where possible:- 
```suggestion
	tflog.Trace(ctx, "deleted deployment", map[string]interface{}{
	    "deployment_id": dResp.UID
	})
```